src/main/java/bio/terra/cli/command/resource/addref/GcsBucketFile.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.cli.command.resource.addref;
+
+import bio.terra.cli.command.shared.BaseCommand;
+import bio.terra.cli.command.shared.options.Format;
+import bio.terra.cli.command.shared.options.ResourceCreation;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.serialization.userfacing.input.CreateGcsBucketFileParams;
+import bio.terra.cli.serialization.userfacing.input.CreateResourceParams;
+import bio.terra.cli.serialization.userfacing.resource.UFGcsBucketFile;
+import bio.terra.workspace.model.StewardshipType;
+import picocli.CommandLine;
+
+/** This class corresponds to the fourth-level "terra resource add-ref gcs-bucket-file" command. */
+@CommandLine.Command(
+    name = "gcs-bucket-file",
+    description = "Add a referenced GCS bucket file.",
+    showDefaultValues = true)
+public class GcsBucketFile extends BaseCommand {
+  @CommandLine.Mixin ResourceCreation resourceCreationOptions;
+
+  @CommandLine.Option(
+      names = "--bucket-name",
+      required = true,
+      description =
+          "Name of the GCS bucket, without the prefix. (e.g. 'my-bucket', not 'gs://my-bucket').")
+  private String bucketName;
+
+  @CommandLine.Option(
+      names = "--file-path",
+      required = true,
+      description = "Full path to the file or folder in the specified GCS bucket.")
```

Comment:
This is where some of my concern comes from, as folders don't really exist and are really an artificial construct of tools (Cloud Console and `gsutil`):

https://cloud.google.com/storage/docs/folders

I'm not sure we would want to take on managing the facade of folders in Terra.  Using "object name" instead of "file path" semantics sidesteps this issue.

Reply:
yes, I'm planning to reuse the file referenced resource endpoint in WSM for folder as well. Nichole is interested in having the --file-path to support folder and files. While folder is not really a thing, WSM will just create a reference as long as the access check pass on the bucket level. In long term, we should have a <string, string> map for random things like this. I discussed with Dan and we are happy to settle with this for now to support the feature request.

Reply:
How can WSM tell if a user is referencing a file or folder? Or it doesn't matter because for both we will just check existence/access to the full bucket?

Reply:
there is a BlobInfo#isDirectory. https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/BlobInfo.html#isDirectory--

But you are right, right now we are just checking the access to the full bucket. I'm putting together a PR to change. this PR will wait till that one is merged.

Reply:
Ah I didn't know about that `isDirectory` method. Will that require `objects.list` permission on the bucket?

Reply:
yeah it does require objects.list premission. .. I don't see how to edit folder permission in google cloud console though so I'm not sure how to verify this.